### PR TITLE
[Bugfix] Cleanup Pixtral HF code

### PR DESCRIPTION
**Pull request by DarkLight1337**
This PR fixes incorrect max image tokens for Pixtral HF, and also cleans up unused code that are already integrated into the merged multi-modal processor.

### CODE REVIEW

**Review by Isotr0py — APPROVED** [review]
Thanks for the cleanup!

File: vllm/model_executor/models/pixtral.py [hunk 1]
```diff
@@ -699,37 +697,14 @@ def get_pixtral_hf_num_patches(*, image_size: int, patch_size: int) -> int:
...
-    return SequenceData.from_prompt_token_counts(
-        (image_token_id, image_feature_size * num_images),
-        (0, seq_len - image_feature_size * num_images),
-    ), {
-        mm_key:
-        consecutive_placeholder_ranges(num_items=num_images,
-                                       item_size=image_feature_size)
-    }
+    # Consider the image_break_token
+    return (grid_length + 1) * grid_length
```
Comment (mgoin) [inline vllm/model_executor/models/pixtral.py:707]: Should this have a plus one at the end for the end token?

**Review by mgoin — COMMENTED** [review]
Thanks for getting rid of some code! Just one question

File: vllm/model_executor/models/pixtral.py (hunk 1 above)
Comment (DarkLight1337) [reply vllm/model_executor/models/pixtral.py:707]: Yes, this is intentional. The reason is that now we include the `image_break_token` in the placeholders.

**Pull request merged**